maint/tuning/coll/json_gen.sh

```diff
@@ -0,0 +1,27 @@
+#! /bin/sh
+
+cfile="src/mpi/coll/src/json_autogen.c"
+
+cat > $cfile<<EOF
+/* -*- Mode: C; c-basic-offset:4 ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+/* DO NOT EDIT!  This file is autogenerated from the corresponding
+ * json files */
+
+EOF
+
+for f in `find maint/tuning/coll -name '*.json'` ; do
+    full=`echo $f | sed -e 's_maint/tuning/coll/__g'`
+    prefix=`echo $full | sed -e 's_/.*__g' | tr a-z A-Z`
+    suffix=`echo $full | sed -e 's_.*/__g' -e 's/\./_/g'`
+    varname="${prefix}_Coll_${suffix}"
+    echo "const char $varname[] = {" >> $cfile
+    cat $f | awk '{ printf("%s ", $0); }' | \
+	sed -e "s/ //g" -e "s/\(.\)/'\1',/g" >> $cfile
```

Comment:
Tab again

Reply:
> I understand that's what you're going for, but I don't see where this code goes to the second option. It looks to me like after you call MPIR_Allgather_intra_brucks, it's going to break out of the switch and then break again out of the for loop to go to fallback and call the auto version of the function. It seems like you need to check mpi_errno to see if the function was successful or not and continue if not.

This particular example doesn't have any algorithms that do not work in specific cases.  That's why a list might not be needed here and the first algorithm will always get picked.  But if you see algorithms like `Iallgatherv`, you'll see algorithms that do not work in specific cases.  In those cases, the specific case would just `continue` and would not hit the `break` till it reaches an algorithm that works.

I added it to all algorithms for consistency, but I can understand if we only want to do this for cases where it's needed for now and have developers add it to the other cases when the need comes up.  What do folks prefer?

Reply:
Ok. I see what you mean now. In that case, I think this is fine.